### ISSUE TITLE
NPE in DeviceUtils

### ISSUE THREAD

**Issue by danikula** [intercom/intercom-android#65]
Version 0.9.5. Stacktrace: 
```
java.lang.RuntimeException: Unable to start receiver com.edufii.push.PushIntentReceiver: java.lang.NullPointerException
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2441)
       at android.app.ActivityThread.access$1700(ActivityThread.java:139)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1286)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5086)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at io.intercom.android.sdk.utilities.DeviceUtils.generateMacId(DeviceUtils.java:62)
       at io.intercom.android.sdk.utilities.DeviceUtils.generateDeviceId(DeviceUtils.java:37)
       at io.intercom.android.sdk.user.DeviceData.generateDeviceData(DeviceData.java:26)
       at io.intercom.android.sdk.api.Api.setGCMPushKey(Api.java:146)
       at io.intercom.android.sdk.Intercom.setupGCM(Intercom.java:490)
       at com.edufii.IntercomHelper.enablePush(IntercomHelper.java:80)
       at com.edufii.push.PushIntentReceiver.onChannelRegistrationSucceeded(PushIntentReceiver.java:31)
       at com.urbanairship.push.BaseIntentReceiver.handleRegistrationIntent(BaseIntentReceiver.java:173)
       at com.urbanairship.push.BaseIntentReceiver.onReceive(BaseIntentReceiver.java:95)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:2434)
       at android.app.ActivityThread.access$1700(ActivityThread.java:139)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1286)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5086)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

**Comment by Harkin** [comment]
We're looking into this one at the moment, did the crash happen on a simulator/emulator or an actual device?

**Comment by danikula** [comment]
Real devices:
GI-I9500_TMMARS , Android 4.1.1
XT1080, Android 4.4.4

**Comment by Harkin** [comment]
:+1: thanks for the info

**Comment by danikula** [comment]
Crash is reproduced on emulators too.

**Comment by DaleCantwell** [comment]
Fixed in 0.9.6
